addons/project/views/project_views.xml

```diff
@@ -512,13 +512,13 @@
                     <templates>
                         <t t-name="kanban-box">
                             <div t-attf-class="#{kanban_color(record.color.raw_value)} oe_kanban_global_click o_has_icon">
-                                <div class="o_project_kanban_main">
+                                <div class="o_project_kanban_main o_text_overflow">
                                     <div class="o_kanban_card_content">
                                         <div class="o_kanban_primary_left">
                                             <div class="o_primary">
                                                 <span><t t-esc="record.name.value"/></span>
                                                 <span t-if="record.partner_id.value">
-                                                    <strong><t t-esc="record.partner_id.value"/></strong>
+                                                    <strong><t class="o_text_overflow" t-esc="record.partner_id.value"/></strong>
```

Comment:
The class should be on the span, not on the `<t/>` tag (so it's not working currently)

You have:
![_tmp_screenshot](https://user-images.githubusercontent.com/54316226/94011046-0e546880-fda7-11ea-9cba-ab710d68601d.png)

Instead of:
![_tmp_screenshot](https://user-images.githubusercontent.com/54316226/94011059-17453a00-fda7-11ea-897e-fae40768e8d6.png)

(you will probably need to set a maximum width to make it work)